services/httpservice/client_test.go

```diff
@@ -115,9 +106,9 @@ func TestDialContextFilter(t *testing.T) {
 		_, err := filter(context.Background(), "", tc.Addr)
 		switch {
 		case tc.IsValid == (err == AddressForbidden) || (err != nil && err != AddressForbidden):
-			t.Errorf("unexpected err for %v (%v)", tc.Addr, err)
+			require.Failf(t, "unexpected err for %v (%v)", tc.Addr, err)
 		case tc.IsValid != didDial:
-			t.Errorf("unexpected didDial for %v", tc.Addr)
+			require.Failf(t, "unexpected didDial for %v", tc.Addr)
 		}
 	}
```

Comment:
@jespino not very sure how I am supposed to refactor this part, could you help

Reply:
I would rewrite the whole thing:

I think is something like this:
```go
if tc.IsValid {
    require.Nil(t, err)
    require.True(t, didDial)
} else {
    require.NotNil(t, err)
    require.Equal(t, err, AddressForbidden)
    require.False(t, didDial)
}
```